src/firefly/java/edu/caltech/ipac/firefly/server/ServCommand.java

```diff
@@ -22,49 +22,39 @@ public void processRequest(HttpServletRequest req, HttpServletResponse res, SrvP
         if (doJsonp) res.setContentType("application/javascript");
 
         String jsonData;
+        JSONObject json=new JSONObject();
+
         try {
             String result = doCommand(new SrvParam(sp.getParamMap()));
 
             if (getCanCreateJson()) {
                 jsonData = result;
             } else {
-                jsonData = "[{" +
-                        "\"success\" :  \"" + true + "\"," +
-                        "\"data\" :  \"" + StringUtils.escapeQuotes(result) + "\"" +
-                        "}]";
-            }
-
-            StringBuilder sb= new StringBuilder(jsonData.length()+10);
-            for(char c : jsonData.toCharArray()) {
-                if (!Character.isISOControl(c) && c>31 && c<127 ) {
-                    sb.append(c);
-                }
+                json.put("success","true" );
+                json.put("data", result );
+                jsonData = "["+json.toJSONString()+"]";
             }
 
-            jsonData= sb.toString();
-
-//            res.setContentType("text/plain");
         } catch (Exception e) {
             e.printStackTrace();
+            json.put("success", "false");
+            json.put("error", e.getMessage());
             StringBuilder sb = new StringBuilder(500);
-            sb.append("[{");
-            sb.append("\"success\" :  \"").append(false).append("\",");
-
-            // need to escape double quotes - otherwise JSON will be messed
-            sb.append("\"error\" : \"").append(e.getMessage().replace("\"", "\\\"")).append("\"");
-
             int cnt = 1;
+            String causeMsg= new String();
             for (Throwable t = e.getCause(); (t != null); t = t.getCause()) {
-                sb.append(", ").append("\"cause").append(cnt).append("\" :  \"").append(t.toString().replace("\"", "\\\"")).append("\"");
+                causeMsg.concat((new Integer(cnt)).toString());
+                causeMsg.concat(":"+t.toString());
                 cnt++;
             }
-            sb.append("}]");
-            jsonData = sb.toString();
-        }
+            json.put("cause", causeMsg);
+            jsonData = "["+json.toJSONString()+"]";
+
+       }
 
         String retval;
         if (doJsonp && callback != null) {
-            retval = callback + "(" + jsonData + ");";
+            retval = callback + jsonData;
```

Comment:
This is most likely wrong.  The enclosed parentheses serve a purpose in JsonP.  Should not remove.

Reply:
We should remove the Jsonp stuff

Reply:
@robyww do you mean remove the code handling jsonp? Can i suggest that we do the cleanup after?